workers/unity/Packages/io.improbable.gdk.core/Utility/Snapshot.cs

```diff
@@ -50,6 +65,11 @@ public EntityId AddEntity(EntityTemplate entityTemplate)
         /// </remarks>
         public void AddEntity(EntityId entityId, EntityTemplate entityTemplate)
         {
+            if (entities.ContainsKey(entityId))
+            {
+                Debug.LogWarning($"Overwriting EntityId {entityId.Id} in Snapshot");
```

Comment:
I'm in half a mind that this should throw an exception since its a destructive action.